src/openfermion/measurements/_probability_distributions.py

```diff
@@ -0,0 +1,363 @@
+            raise ValueError('''Prior distribution of phases has shape {},
+                required shape is {}'''.format(
+                    fourier_vector.shape, (2*num_freqs+1, num_vectors)))
+
+        if amplitude_mean is None:
+            amplitude_mean = [1]
+        if amplitude_var is None:
+            amplitude_var = [[EQ_TOLERANCE]]
+
+        if num_vectors != len(amplitude_mean):
+            raise ValueError('''I need {} amplitudes, but you have
+                given me {}.'''.format(num_vectors, len(amplitude_mean)))
+
+        if any([numpy.isclose(amplitude_mean[j],
+                              amplitude_mean[j+1])
+                for j in range(num_vectors - 1)]):
+            raise ValueError('''Starting amplitudes must be different.''')
+
+        # The amplitude distributions are not being updated at this point
+        # (instead the data for the update is stored in p_vecs). As such,
+        # we just store this as a prior.
+        self._prior_amplitude_mean = amplitude_mean
+        self._prior_amplitude_var = amplitude_var
+
+        # The current estimate of the amplitudes is just the mean
+        # of the given guess.
+        self._amplitude_estimates = numpy.array(amplitude_mean)
+
+        if fourier_vector is not None:
+            self._fourier_vectors = numpy.array(fourier_vector)
+        else:
+            # Initialize probability distributions
+            self._fourier_vectors = numpy.zeros((self._num_freqs*2+1,
+                                                 self._num_vectors))
+            for j in range(self._num_vectors):
+                self._fourier_vectors[0, j] = 1
+
+        # This stores data from repeated Bayesian updates in order to
+        # calculate data of the updated distribution as required.
+        self._p_vecs = []
+
+        self._inv_covar_mat = numpy.linalg.inv(
+            numpy.array(self._prior_amplitude_var))
+
+    def get_real_dist(self, x_vec=None):
+        """
+        Generates the real distribution over a sequence of points
+
+        Args:
+            x_vec (numpy array or list): the points at which the distribution is generated.
+                If None, defaults to 101 points between -pi and pi.
+        Returns:
+            y_vecs (numpy array): the value of each distribution at the given x points.
+                y_vecs[j,k] = f_j(x_vec[k])
+        """
+        # Set default x vector
+        if x_vec is None:
+            x_vec = numpy.linspace(-pi, pi, 101)
+
+        # Init storage to return
+        y_vecs = []
+
+        for j in range(self._num_vectors):
+
+            # Get vector and evaluate corresponding function
+            # at the chosen set of points
+            vector = self._fourier_vectors[:, j]
+            y_vec = [self._make_dist(vector, theta) for theta in x_vec]
+
+            # Store and plot data
+            y_vecs.append(y_vec)
+
+        return numpy.array(y_vecs)
+
+    def _holevo_centers(self, vectors=None):
```

Comment:
This hidden method isn't used anywhere in the class.

Reply:
Very good point, this shouldn't be hidden.